### PR TITLE
changelog for 0.5.0

### PR DESCRIPTION
**Pull request by psychocoderHPC**
@ComputationalRadiationPhysics/picongpu-maintainers @ComputationalRadiationPhysics/picongpu-developers please check the change log and suggest updates for the short summery of this release.

I created the change-log with our change-log tool (thx @ax3l ) and did some manual corrections.

The contributor list was generated with the following command and was manually adjusted.

```
git log d340812641320d3a61e343a521287fd0720ba55b..HEAD | grep Author | sort | uniq | sed "s/ \+/ /g; s/^ //" | cut -d" " -f2-3
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,277 @@
...
+
+**Date:** 2020-03-XX
+
+Perfectly Matched Layer (PML) and Bug Fixes
+
+This release introduce PML boundary conditions for the Yee field solver and particle 
+rendering with the ISAAC plugin.
+
+Thanks to Igor Andriyash, Sergei Bastrakov, Xeinia Bastrakova, Andrei Berceanu,
+Finn-Ole Carstens, Alexander Debus, Jian Fuh, Marco Garten, Axel Huebl,
```
Comment (n01r) [inline CHANGELOG.md]: > Jian Fuh

His full name is `Jian Fuh Ong`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,277 @@
...
 =========
 
+0.5.0
+-----
+
+**Date:** 2020-03-XX
+
+Perfectly Matched Layer (PML) and Bug Fixes
+
+This release introduce PML boundary conditions for the Yee field solver and particle
```
Comment (n01r) [inline CHANGELOG.md]: _introduces_

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,277 @@
...
+**Date:** 2020-03-XX
+
+Perfectly Matched Layer (PML) and Bug Fixes
+
+This release introduce PML boundary conditions for the Yee field solver and particle 
+rendering with the ISAAC plugin.
+
+Thanks to Igor Andriyash, Sergei Bastrakov, Xeinia Bastrakova, Andrei Berceanu,
+Finn-Ole Carstens, Alexander Debus, Jian Fuh, Marco Garten, Axel Huebl,
+Sophie Kossag, Anton Lebedev, Felix Meyer, Pawel Ordyna, Richard Pausch,
```
Comment (n01r) [inline CHANGELOG.md]: `Koßagk` but now she's `Sophie Rudat`. How do we want to handle new names?
Just update them or write `Sophie Rudat (Koßagk)`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,277 @@
...
+   - OpenMPI: Use ROMIO for IO #2857
+   - Radiation Plugin: fix bool conditions for hdf5 output #3021
+   - CMake Modules: Update ADIOS FindModule #3116
+   - ADIOS Particle Writer: Fix timeOffset #3120
+   - openPMD: use particle id naming #3165
+   - Include int16 and uint16 types as traits for ADIOS #2929
+   - Fix observation direction of transition radiation plugin #3091
+   - Fix doc transition radiation plugin #3089
+   - Fix doc rad plugin units and factors #3113
+   - Fix wrong underline in TrasRad plugin doc #3102
```
Comment (n01r) [inline CHANGELOG.md]: _TransRad_ 

I fixed the title of the pull just so that it is searchable without the typo even though the content of the pull is minor

File: CHANGELOG.md (hunk 3 above)
Comment (psychocoderHPC) [reply CHANGELOG.md]: I will keep Koßagk because here github account is showing this name.

**Comment by ax3l** [comment]
@psychocoderHPC Just FYI: we previously created the ordering of contributors by number of PRs like this, https://github.com/ComputationalRadiationPhysics/picongpu/wiki/Maintainer-Documentation, but alphabetical is a good idea as well.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,277 @@
...
+**Date:** 2020-03-XX
+
+Perfectly Matched Layer (PML) and Bug Fixes
+
+This release introduces PML boundary conditions for the Yee field solver and particle 
+rendering with the ISAAC plugin.
+
+Thanks to Igor Andriyash, Sergei Bastrakov, Xeinia Bastrakova, Andrei Berceanu,
+Finn-Ole Carstens, Alexander Debus, Jian Fuh Ong, Marco Garten, Axel Huebl,
+Sophie Koßagk, Anton Lebedev, Felix Meyer, Pawel Ordyna, Richard Pausch,
```
Comment (ax3l) [inline CHANGELOG.md]: Sophie married two years ago and is now Sophie Rudat :)
```suggestion
Sophie Rudat, Anton Lebedev, Felix Meyer, Pawel Ordyna, Richard Pausch,
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,277 @@
...
+ - Particle pusher acceleration #2731
+ - stop moving window after N steps #2792
+ - Remove unused ABSORBER_FADE_IN_STEPS from .param files in examples #2942
+ - add namespace "radiation" around code related to radiation plugin #3004
+ - Add a runtime parameter for window move point #3022
+ - Ionization: add silicon to pre-defines #3078
+ - Make dependency between boundElectrons and atomicNumbers more explicit #3076
+ - openPMD: use particle id naming #3165
+ - Fix species.param docu (in all examples too) #2795
+ - Fix species.param example doc and grammar #2793
```
Comment (ax3l) [inline CHANGELOG.md]:
```suggestion
 - Docs: update `species.param` #2793 #2795
```

File: docs/propose_changelog.py [hunk 7]
```diff
@@ -13,7 +13,7 @@
 
 # config
 #          see: https://github.com/settings/tokens
-g = Github("add token with public repo read rights here")
+g = Github("49265fe2d70063570d92fbabc0c966837d0062bc")
```
Comment (ax3l) [inline docs/propose_changelog.py]: uhoh... this please:
- revert this change
- quickly remove this token in your settings, this is a security leak with access to your account:
https://github.com/settings/tokens

Comment (psychocoderHPC) [reply docs/propose_changelog.py]: thx deleted the token. I will do an update for this tool to read the token from a file in home.

Comment (ax3l) [reply docs/propose_changelog.py]: update: token deleted

File: CHANGELOG.md (hunk 5 above)
Comment (n01r) [reply CHANGELOG.md]: Yea, already commented that but resolved it when @psychocoderHPC said that you can find her under that name on GitHub.
Ideally, you'd also want to find someone's work by their previous name and their current work and contact by their new name.
So `<first name> <new name> (<old name>)`  ... but IDK, that's how I would do it.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,277 @@
...
+   - fix wrong constructor call #3117
+   - Fix calculation of omega_p for logging #3163
+   - Fix laser bug in case focus position is at the init plane #2922
+   - Fix binomial current interpolation #2838
+   - Fix particle creation if density zero #2831
+   - Avoid two slides #2774
+   - Fix warning: comparison of unsigned integer #2987
+   - Fix style issues with particlePusherAcceleration #2781
+   - Fix species.param docu (in all examples too) #2795
+   - Fix species.param example doc and grammar #2793
```
[the remark this answers is not in the document]
Comment (ax3l) [reply CHANGELOG.md]: A documentation fix can be a bug in some cases.

**Comment by ax3l** [comment]
Thank you for preparing the changelog, @psychocoderHPC! Looks great!

File: CHANGELOG.md (hunk 5 above)
Comment (psychocoderHPC) [reply CHANGELOG.md]: @n01r is trying to get in contact with here.

Comment (n01r) [reply CHANGELOG.md]: She agrees with the suggestion `Sophie Rudat (Koßagk)`

**Comment by psychocoderHPC** [comment]
PR ready to merge

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,6 +1,276 @@
...
+**Date:** 2020-03-XX
+
+Perfectly Matched Layer (PML) and Bug Fixes
+
+This release introduces PML boundary conditions for the Yee field solver and particle 
+rendering with the ISAAC plugin.
+
+Thanks to Igor Andriyash, Sergei Bastrakov, Xeinia Bastrakova, Andrei Berceanu,
+Finn-Ole Carstens, Alexander Debus, Jian Fuh Ong, Marco Garten, Axel Huebl,
+Sophie Tudat (Koßagk), Anton Lebedev, Felix Meyer, Pawel Ordyna, Richard Pausch,
```
Comment (sbastrakov) [inline CHANGELOG.md]: Now there is a typo in Sophie's first name.

**Comment by psychocoderHPC** [comment]
> A typo.
> 
> Also, do you want me to add a sentence about PML?

If you mean the introduction? Yes feel free to provide more description via PR or push directly to my branch.
The bugs https://github.com/ComputationalRadiationPhysics/picongpu/pull/3216 do not need to be mentioned because the PML is new and there are no changes/bug compared to the last release but you can add the PR numbers to the point where we mentioned the new implementation of PML.

**Comment by psychocoderHPC** [comment]
@sbastrakov ping, please add the PML sentence you wrote about.

**Comment by sbastrakov** [comment]
Thanks for pinging @psychocoderHPC . I made a PR to your fork with PML and other suggested extensions to the text.

**Comment by psychocoderHPC** [comment]
please merge

**Comment by PrometheusPi** [comment]
Still looks good to me - will merge.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,6 +1,287 @@
...
+   - Add an option to account for PML in the memory calculator #3029
+   - Update profile hemera-hzdr: CMake version #3059
+   - Travis CI: OSX sed Support #3073
+   - CMake: mark cuda 10.2 as tested #3118
+   - Avoid bash completion file path repetition #3136
+   - Bashcompletion #3069
+   - Jupyter widgets output capture #3149
+   - Docs: Add ionization prediction plot #2870
+   - pic-edit: clean cmake file cache if new param added #2904
+   - CMake: Honor _ROOT Env Hints #2891
```
Comment (ax3l) [inline CHANGELOG.md:87]: Formatting suggstion: for the next changelog update in the release branch, maybe format as `_ROOT` here (see issue in GitHub preview)

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,6 +1,287 @@
...
+     - Add namespace "radiation" around code related to radiation plugin #3004
+     - Remove unused includes of pthread #3040
+     - SpeciesEligibleForSolver for radiation plugin #3061
+     - ADIOS: Avoid unsafe temporary strings #2946
+   - tools:
+     - Update cuda_memtest: CMake CUDA_ROOT Env #2892
+     - Update hemera tpl after SLURM update  #3123
+   - Add pillow as dependency #3180
+   - Params: remove `boost::vector<>` usage #2769
+   - Use _X syntax in OnceIonized manipulator #2745
```
Comment (ax3l) [inline CHANGELOG.md:204]: @psychocoderHPC here as well: `_X` :)

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,6 +1,287 @@
...
+   - Doc: Add System Links #2818
+   - remove grep file redirect #2788
+   - Correct jupyter widget example #3191
+   - fix typo: `UNIT_LENGHT` to `UNIT_LENGTH` #3194
+   - Change link to CRP group @ HZDR #2814
+ - Examples: Unify .cfg #2826
+ - Remove unused ABSORBER_FADE_IN_STEPS from .param files in examples #2942
+ - Field absorber test example #2948
+ - Singularity: Avoid Dotfiles in Home #2981
+ - Boost: No std::auto_ptr #3012
```
Comment (ax3l) [inline CHANGELOG.md:271]: and here: `std::auto_ptr`
